server/lib/mutations/command_errors.rb

```diff
@@ -0,0 +1,47 @@
+require 'mutations'
+
+module Mutations
+  class Command
+    protected
+
+    # Add error for a key
+    #
+    # @param key [Symbol] :foo
+    # @param key [String] 'foo.bar'
+    # @param error [Symbol] :not_found
+    # @param error [Mutations:ErrorAtom, Mutations::ErrorArray, Mutations::ErrorHash]
+    def add_error(key, error, message = nil)
+      if error.is_a? Symbol
+        error = ErrorAtom.new(key, error, message: message)
+      elsif error.is_a?(Mutations::ErrorAtom) || error.is_a?(Mutations::ErrorArray) || error.is_a?(Mutations::ErrorHash)
+
+      else
+        raise ArgumentError.new("Invalid error of kind #{error.class}")
+      end
+
+      @errors ||= ErrorHash.new
+      @errors.tap do |errs|
+        path = key.to_s.split(".")
+        last = path.pop
+        inner = path.inject(errs) do |cur_errors,part|
+          cur_errors[part.to_sym] ||= ErrorHash.new
+        end
+        inner[last] = error
```

Comment:
This is otherwise exact copy-pasta from [`Mutations::Command#add_error`](https://github.com/cypriss/mutations/blob/v0.7.2/lib/mutations/command.rb#L122), except this line is changed
```ruby
 inner[last] = ErrorAtom.new(key, kind, :message => message)
```